src/wbase/SendChannelShared.cc

```diff
@@ -247,37 +242,41 @@ bool SendChannelShared::_transmit(bool erred) {
     return true;
 }
 
-util::TimerHistogram transmitHisto("transmit Hist", {0.1, 1, 5, 10, 20, 40});
-
 bool SendChannelShared::_sendBuf(lock_guard<mutex> const& streamLock, xrdsvc::StreamBuffer::Ptr& streamBuf,
                                  bool last, string const& note, int scsSeq) {
     bool sent = _sendChannel->sendStream(streamBuf, last, scsSeq);
     if (!sent) {
         LOGS(_log, LOG_LVL_ERROR, "Failed to transmit " << note << "!");
         return false;
     } else {
-        util::Timer t;
-        t.start();
         LOGS(_log, LOG_LVL_INFO, "_sendbuf wait start " << note);
-        streamBuf->waitForDoneWithThis();  // Block until this buffer has been sent.
-        t.stop();
-        auto logMsg = transmitHisto.addTime(t.getElapsed(), note);
-        LOGS(_log, LOG_LVL_DEBUG, logMsg);
+        // Only wait if this is really the last message to finish transmitting for this SendChannelShared.
+        // Note: If this does speed up transmission of data to the czar then it is probably worth it to
+        //       add code to wait if more than X buffers are in use at the same time, which will likely
+        //       make a big performance crater, but better than crashing due to out of memory.
+        //       It may be best to reduce the number of threads per scheduler to 1 until it clears up instead
+        //       of waiting for every send.
+        //       If this doesn't improve transmission performance, just change it back to waiting all of the
+        //       time.
+        if (last) {
+            streamBuf->waitForDoneWithThis();  // Block until this buffer has been sent.
```

Comment:
This is actually a very big change to the behavior of Qserv workers. It needs to be in a separate JIRA ticket (and PR) to track and isolate possible side effects of the change. My understanding is that the declared scope of the present PR is just to streamline statistic collection and Qserv monitoring as in:
```
Moved timers and statistics functions to where they now need to be
```

Please, either remove this from this PR or redefine the scope of the JIRA ticket and the PR to reflect actual changes in the code.

Reply:
This is the change that forced most of the other changes. Moving it to another ticket is pretty simple.